### ISSUE TITLE
BehaviorSubject does not seem to use its initialValue (RC3, scala.js)

### ISSUE THREAD

**Issue by niladic** [monifu/monifu#83]
I was trying to upgrade a Scala.js project from 1.0-M2 to 1.0-RC3, and it seems the semantics of `BehaviorSubject` have changed between the two versions. The following would print the initialValue on M2, but does not on RC3:
```scala
val subject = monifu.reactive.subjects.BehaviorSubject[Int](10)
subject.foreach(println)
```

My use case is actually to transform an `Observable` into a state machine which can always give its current state (even when the source has not emitted a first element). For example let's say I have some `Observable` of clicks and I want a shared switch:

```scala
val clicks = PublishSubject[Unit]()

val switch: Observable[Boolean] = {
  val connectable = clicks.scan[Boolean](false) { case (old, _) => !old }.behavior(false)
  connectable.connect()
  connectable
}
```

It is possible to simply use a `BehaviorChannel` and `pushNext` the initial state. But had I the semantics of `BehaviorSubject` wrong or is it is a bug ?

**Comment by alexandru** [comment]
Yes, the semantics have changed, however this will get reverted. Thanks for reporting it and please have patience while I fix it. Will happen sometime this week.

**Comment by niladic** [comment]
Ok thanks !

**Comment by mseddon** [comment]
As an aside- this issue appears to have started in `M11`, so until this fix is released, users can still upgrade to `M10`.

**Comment by sikor** [comment]
There is also a problem in Scala version (not scala.js). It occurs when source Observable for behavior subject is not completed. The first code snippet does not print anything but second works ok. 
``` scala
  test("Behavior with uncompleted source") {
    val obs = PublishChannel[String](Unbounded).behavior("init")
    obs.connect()
    obs.foreach(s => println(s)) // expected to print "init" but prints nothing. 
  }
```

```scala
  test("behavior with completed source") { 
    val obs = Observable().behavior("init")
    obs.connect()
    obs.foreach(s => println(s)) // prints "init"
  }
```

**Comment by alexandru** [comment]
@sikor, @mseddon, @niladic - guys, I'm sorry for keeping you waiting. This issue is now fixed.

The history of this problem is that I've been very enthusiastic about my new way of synchronizing subjects in a wait-free fashion that I've broken the contract of `BehaviorSubject` and of `ReplaySubject` without noticing it. Should I ever break this contract again, it should be caught by tests.

The behavior of `BehaviorSubject` and that of `ReplaySubject` is now fixed and should work both on top of the JVM and on top of JS.  I have released version `1.0-M4` of Monifu. 

Please try it out and let me if you are experiencing any more problems.

**Comment by mseddon** [comment]
Thanks @alexandru! I will check this out tomorrow.  I know how it is to be dragged away by work :)
